src/nxt_fs.c

```diff
@@ -63,12 +65,23 @@ nxt_fs_mkdir_parent(const u_char *path, mode_t mode)
     ret = NXT_OK;
 
     ptr = strrchr(dir, '/');
-    if (nxt_slow_path(ptr == NULL || ptr == dir)) {
+    if (ptr == dir || nxt_slow_path(ptr == NULL)) {
         goto out_free;
     }
 
     *ptr = '\0';
+    if (strcmp(dir, (const char *) path) != 0)
+    {
+        ret = nxt_fs_mkdir_parent((const u_char *) dir, mode);
+        if (nxt_slow_path(ret == NXT_ERROR)) {
+            goto out_free;
+        }
+    }
+
     ret = nxt_fs_mkdir((const u_char *) dir, mode);
+    if (ret == NXT_ERROR && errno == EEXIST) {
+        ret = NXT_OK;
+    }
 
 out_free:
     nxt_free(dir);
```

Comment:
Didn't you just create 'mkdir -p'?

Maybe the function should be renamed more appropriately (assuming it doesn't need to be retained in its original form)

Also not terribly fond of the recursion... or if some nutter tries to create `/a/b/c/d/e/f/g/h/i/j/...`, but I can probably live with it...

Reply:
Yeah, I thought of looping, but since this is something we run just twice, and only at program startup, I guess it's simpler to recurse.

Regarding the name...  I thought originally of naming it _parents instead of _parent.  However, I thought _parents might be confusing, as it could lead to think that it's equivalent to `mkdir -p`, while it's equivalent to `mkdir -p $(dirname ...)`.  Now that I write this, the obvious name comes to my head as *nxt_fs_mkdir_dirname_parents*().  Does that sound good to you?

Reply:
Or maybe _mkdir_parents_dirname ...